ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -69,6 +81,17 @@ export const ExportDatasetModal: (props: Props) => JSX.Element = fp.flow(withCur
       }
     }
 
+    function createExportDatasetRequest(): DataSetExportRequest {
```

Comment:
Existing issue, but why use this function declaration style within this function component? I think it would be more typical to use an arrow function and a constant for these kinds of closurized anonymous functions. The function keyword has some special implications which I'm not sure are desired in this situation.

Alternatively - is there some special reason to use the `function` keyword here that I'm missing?

Possibly this was just carried over from the conversion to hooks. e.g. I would have expected:
```
const createExportDatasetRequest = () => {
  ...
};
```

Reply:
I actually did choose to use the function syntax but my stance on this can be changed - I was wavering a bit when I made the decision but I went with it for two reasons

When quickly skimming through code, I can more quickly pick out what's a function versus a value by seeing the function keyword. Ex function createExportDatasetRequest() {} vs const createExportDatasetRequest = () => {}
I like not having to worry about the order of declaring functions the way it's not really a concern when writing class methods in Java.

I'm going to keep it as is for now but would like to chat about it when we're back from vacation.